src/webgpu/shader/execution/builtin/abs.spec.ts

```diff
@@ -0,0 +1,301 @@
+      .combineWithParams([{ baseType: 'i32' }] as const)
+      .beginSubcases()
+      .expandWithParams(generateTypes)
+  )
+  .fn(async t => {
+    assert(t.params._kTypeInfo !== undefined, 'generated type is undefined');
+    runShaderTest(
+      t,
+      t.params.storageClass,
+      t.params.storageMode,
+      t.params.baseType,
+      t.params.type,
+      t.params._kTypeInfo.arrayLength,
+      'abs',
+      OperandType.Int,
+      [
+        { input: -1, expected: [1] },
+        { input: -2, expected: [2] },
+        { input: -4, expected: [4] },
+        { input: -8, expected: [8] },
+        { input: -16, expected: [16] },
+        { input: -32, expected: [32] },
+        { input: -64, expected: [64] },
+        { input: -128, expected: [128] },
+        { input: -256, expected: [256] },
+        { input: -512, expected: [512] },
+        { input: -1024, expected: [1024] },
+        { input: -2048, expected: [2048] },
+        { input: -4096, expected: [4096] },
+        { input: -8192, expected: [8192] },
+        { input: -16384, expected: [16384] },
+        { input: -32768, expected: [32768] },
+        { input: -65536, expected: [65536] },
+        { input: -131072, expected: [131072] },
+        { input: -262144, expected: [262144] },
+        { input: -524288, expected: [524288] },
+        { input: -1048576, expected: [1048576] },
+        { input: -8388607, expected: [8388607] },
+        { input: -16777215, expected: [16777215] },
+        { input: -16777216, expected: [16777216] },
+        { input: -134217727, expected: [134217727] },
+        { input: -1073741823, expected: [1073741823] },
+        { input: -2147483647, expected: [2147483647] },
+        { input: 0, expected: [0] },
+        { input: 1, expected: [1] },
+        { input: 2, expected: [2] },
+        { input: 4, expected: [4] },
+        { input: 8, expected: [8] },
+        { input: 16, expected: [16] },
+        { input: 32, expected: [32] },
+        { input: 64, expected: [64] },
+        { input: 128, expected: [128] },
+        { input: 256, expected: [256] },
+        { input: 512, expected: [512] },
+        { input: 1024, expected: [1024] },
+        { input: 2048, expected: [2048] },
+        { input: 4096, expected: [4096] },
+        { input: 8192, expected: [8192] },
+        { input: 16384, expected: [16384] },
+        { input: 32768, expected: [32768] },
+        { input: 65536, expected: [65536] },
+        { input: 131072, expected: [131072] },
+        { input: 262144, expected: [262144] },
+        { input: 524288, expected: [524288] },
+        { input: 1048576, expected: [1048576] },
+        { input: 8388607, expected: [8388607] }, //2^23 - 1
+        { input: 16777215, expected: [16777215] }, //2^24 - 1
+        { input: 16777216, expected: [16777216] }, //2^24
+        { input: 134217727, expected: [134217727] }, //2^27 - 1
+        { input: 1073741823, expected: [1073741823] }, //2^30 - 1
```

Comment:
nit: Case for 2147483647?

also nit: Can write cases as expressions or `0x` literals as it might be clearer. E.g. `0xFFFF_FFFF` or `-0x1_0000_0000`

Reply:
next cl